### PR TITLE
Add external volume to hazelcast StatefulSet

### PR DESCRIPTION
**Pull request by galibey**
This is another setting to mount volume to hazelcast instance pod. Unlike `customVolume` it is not added to hz classpath. 

One of the use-cases is to configure diagnostic logs and heap dumps to be stored to `externalVolume` for further collection by another service.

PR for 3.12.x version will come after reviews passed here.

### CODE REVIEW

File: stable/hazelcast-enterprise/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: hazelcast-enterprise
-version: 3.5.8
+version: 3.5.9
```
Comment (leszko) [inline stable/hazelcast-enterprise/Chart.yaml:3]: I think we can increase the "minor" version (not the "patch" version).

**Review by leszko — APPROVED** [review]
Thanks for the change @galibey . Added one minor comment. Other than that, it looks good to me. But please wait for the review from @hasancelik before merging.

File: stable/hazelcast-enterprise/Chart.yaml (hunk 1 above)
Comment (galibey) [reply stable/hazelcast-enterprise/Chart.yaml]: Done

**Review by hasancelik — APPROVED** [review]
Thanks for the change @galibey 👍

**Pull request merged**